tests/func-tests/virt-launcher-mutating-webhook.go

```diff
@@ -0,0 +1,163 @@
+	"kubevirt.io/kubevirt/tests/flags"
+	kvtutil "kubevirt.io/kubevirt/tests/util"
+
+	g "github.com/onsi/ginkgo/v2"
+)
+
+var _ = g.Describe("virt-launcher mutating webhook", func() {
+	flag.Parse()
+
+	var hcoOriginalAnnotations map[string]string
+
+	g.BeforeEach(func() {
+		BeforeEach()
+
+		hco := getHco()
+		hcoOriginalAnnotations = hco.Annotations
+	})
+
+	g.AfterEach(func() {
+		hco := getHco()
+		hco.Annotations = hcoOriginalAnnotations
+		hcoOriginalAnnotations = map[string]string{}
+
+		setHco(hco)
+	})
+
+	g.Context("guest-to-request memory headroom", func() {
+
+		enableWebhook := func(customHeadroom *string) {
+			hco := getHco()
+			if hco.Annotations == nil {
+				hco.Annotations = map[string]string{}
+			}
+
+			hco.Annotations["kubevirt.io/enable-guest-to-request-memory-headroom"] = "true"
+			if customHeadroom != nil {
+				hco.Annotations["kubevirt.io/custom-guest-to-request-memory-headroom"] = *customHeadroom
+			}
+
+			setHco(hco)
+		}
+
+		g.It("with default headroom", func() {
+			enableWebhook(nil)
+
+			vmi := kvtests.NewRandomVMI()
+			vmi = kvtests.RunVMIAndExpectLaunch(vmi, 60)
+			launcherPod := kvtests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
+
+			vmiMemRequest := vmi.Spec.Domain.Resources.Requests[v1.ResourceMemory]
+
+			expectedPodRequest := vmiMemRequest.DeepCopy()
+			expectedPodRequest.Add(resource.MustParse("2G"))
+
+			actualPodRequest := getComputeContainer(launcherPod).Resources.Requests[v1.ResourceMemory]
+
+			Expect(actualPodRequest.Cmp(expectedPodRequest)).To(BeTrue(), fmt.Sprintf("expecting pod requests (%v) to equal %v", actualPodRequest, expectedPodRequest))
+		})
+
+	})
+
+})
+
+func getHco() *hcov1beta1.HyperConverged {
+	virtCli, err := kubecli.GetKubevirtClient()
+	Expect(err).ToNot(HaveOccurred())
+
+	SkipIfNotOpenShift(virtCli, "DataImportCronTemplate")
+
+	cli, err := kubecli.GetKubevirtClientFromRESTConfig(virtCli.Config())
+	Expect(err).ToNot(HaveOccurred())
+
+	hc := &hcov1beta1.HyperConverged{}
+	Expect(cli.RestClient().
+		Get().
+		Resource("hyperconvergeds").
+		Name("kubevirt-hyperconverged").
+		Namespace(flags.KubeVirtInstallNamespace).
+		AbsPath("/apis", hcov1beta1.SchemeGroupVersion.Group, hcov1beta1.SchemeGroupVersion.Version).
+		Timeout(10 * time.Second).
+		Do(context.TODO()).
+		Into(hc),
+	).To(Succeed())
+
+	return hc
+}
+
+func setHco(hc *hcov1beta1.HyperConverged) *hcov1beta1.HyperConverged {
+	virtCli, err := kubecli.GetKubevirtClient()
+	Expect(err).ToNot(HaveOccurred())
+
+	SkipIfNotOpenShift(virtCli, "DataImportCronTemplate")
+
+	cli, err := kubecli.GetKubevirtClientFromRESTConfig(virtCli.Config())
+	Expect(err).ToNot(HaveOccurred())
+
+	res := cli.RestClient().Put().
+		Resource("hyperconvergeds").
+		Name(hcov1beta1.HyperConvergedName).
+		Namespace(flags.KubeVirtInstallNamespace).
+		AbsPath("/apis", hcov1beta1.SchemeGroupVersion.Group, hcov1beta1.SchemeGroupVersion.Version).
+		Timeout(10 * time.Second).
+		Body(hc).Do(context.TODO())
```

Comment:
This is failing for not having the kind and the apiVersion fields in the hyperConverged CR. 

You can add them just above these lines; e.g.
```suggestion
	hc.APIVersion = "hco.kubevirt.io/v1beta1"
	hc.Kind = "HyperConverged"

	res := cli.RestClient().Put().
		Resource("hyperconvergeds").
		Name(hcov1beta1.HyperConvergedName).
		Namespace(flags.KubeVirtInstallNamespace).
		AbsPath("/apis", hcov1beta1.SchemeGroupVersion.Group, hcov1beta1.SchemeGroupVersion.Version).
		Timeout(10 * time.Second).
		Body(hc).Do(context.TODO())
```